core/camel-base-engine/src/main/java/org/apache/camel/impl/engine/DefaultInflightRepository.java

```diff
@@ -67,28 +67,28 @@ public void remove(Exchange exchange) {
 
     @Override
     public void add(Exchange exchange, String routeId) {
-        AtomicInteger existing = routeCount.get(routeId);
+        LongAdder existing = routeCount.get(routeId);
         if (existing != null) {
-            existing.incrementAndGet();
+            existing.increment();
         }
     }
 
     @Override
     public void remove(Exchange exchange, String routeId) {
-        AtomicInteger existing = routeCount.get(routeId);
+        LongAdder existing = routeCount.get(routeId);
         if (existing != null) {
-            existing.decrementAndGet();
+            existing.increment();
```

Comment:
@davsclaus i think the bug could be here. It should decrement, but it is incrementing.